registry/remote/repository.go

```diff
@@ -361,38 +409,92 @@ func (r *Repository) referrers(ctx context.Context, artifactType string, fn func
 	if resp.StatusCode != http.StatusOK {
 		return "", errutil.ParseErrorResponse(resp)
 	}
-	if !legacyAPI {
-		if err := verifyOrasApiVersion(resp); err != nil {
-			return "", err
-		}
-	}
 
-	var page struct {
-		References []ocispec.Descriptor `json:"references"`
-		Referrers  []ocispec.Descriptor `json:"referrers"`
-	}
+	var index ocispec.Index
 	lr := limitReader(resp.Body, r.MaxMetadataBytes)
-	if err := json.NewDecoder(lr).Decode(&page); err != nil {
+	if err := json.NewDecoder(lr).Decode(&index); err != nil {
 		return "", fmt.Errorf("%s %q: failed to decode response: %w", resp.Request.Method, resp.Request.URL, err)
 	}
-	var refs []ocispec.Descriptor
-	if legacyAPI {
-		refs = page.References
-	} else {
-		refs = page.Referrers
+	referrers := index.Manifests
+	if artifactType != "" &&
+		!isReferrersFilterApplied(index.Annotations[ocispec.AnnotationReferrersFiltersApplied], "artifactType") {
+		// perform client side filtering if the filter is not applied on the
+		// server side
```

Comment:
I guess the comment length limitation should not applied to this line since it will not be converted to any documentation?

Reply:
Merged into one line.